src/schemas/global.ts

```diff
@@ -1,57 +1,69 @@
 // we need this file used in classic json schema
 // duplicate of shared fluent-schema.ts
+import { Type } from '@sinclair/typebox';
+import { StatusCodes } from 'http-status-codes';
+
+import { customType, registerSchemaAsRef } from '../plugins/typebox';
 
 export const UUID_REGEX =
   '^[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12}$';
 export const NAME_REGEX = '^\\S+( \\S+)*$';
 // allow empty strings or words separated by one space
 export const EMPTY_OR_SPACED_WORDS_REGEX = /^(\S+( \S+)*)?$/;
 
-export default {
-  $id: 'https://graasp.org/',
-  definitions: {
-    uuid: {
-      type: 'string',
-      pattern: UUID_REGEX,
-    },
-    itemPath: {
-      type: 'string',
-      pattern:
-        '^[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12}' +
-        '(.[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12})*$',
+export const entityIdSchemaRef = registerSchemaAsRef(
+  Type.Object(
+    {
+      // Object Definition
+      id: customType.UUID(),
     },
-    idParam: {
-      type: 'object',
-      required: ['id'],
-      properties: {
-        id: { $ref: '#/definitions/uuid' },
-      },
+    {
+      // Schema Options
+      title: 'Entity ID',
+      $id: 'entityId',
       additionalProperties: false,
     },
-    idsQuery: {
-      type: 'object',
-      required: ['id'],
-      properties: {
-        id: {
-          type: 'array',
-          items: { $ref: '#/definitions/uuid' },
-          uniqueItems: true,
-        },
-      },
+  ),
+);
+
+export const idsQuerySchemaRef = registerSchemaAsRef(
+  Type.Object(
+    {
+      // Object Definition
+      id: Type.Array(customType.UUID(), { uniqueItems: true }),
+    },
+    {
+      // Schema Options
+      title: 'IDs Query',
+      $id: 'idsQuery',
       additionalProperties: false,
     },
-    error: {
-      type: 'object',
-      properties: {
-        name: { type: 'string' },
-        code: { type: 'string' },
-        message: { type: 'string' },
-        statusCode: { type: 'number' },
-        data: {},
-        origin: { type: 'string' },
-        // stack: {}
-      },
+  ),
+);
```

Comment:
Do you think it is necessary?

Reply:
I don't think it is necessary, I can remove it for sure !